### PR TITLE
Url encode the web dav url for copying

### PR DESCRIPTION
**Pull request by miaulalala**

### CODE REVIEW

File: apps/files/templates/appnavigation.php [hunk 1]
```diff
@@ -48,7 +48,7 @@
 			</div>
 			<label for="webdavurl"><?php p($l->t('WebDAV')); ?></label>
 			<input id="webdavurl" type="text" readonly="readonly"
-				   value="<?php p($_['webdav_url']); ?>"/>
+				   value="<?php p(urlencode($_['webdav_url'])); ?>"/>
```
Comment (nickvergessen) [inline apps/files/templates/appnavigation.php:51]: You actually want to urlencode the parameter being added into route

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Review by szaimen — COMMENTED** [review]
also a test would be good, I guess

File: apps/files/lib/Controller/ViewController.php [hunk 2]
```diff
@@ -266,7 +266,7 @@ public function index($dir = '', $view = '', $fileid = null, $fileNotFound = fal
 		$nav->assign('quota', $storageInfo['quota']);
 		$nav->assign('usage_relative', $storageInfo['relative']);
 
-		$nav->assign('webdav_url', \OCP\Util::linkToRemote('dav/files/' . $user));
+		$nav->assign('webdav_url', \OCP\Util::linkToRemote('dav/files/' . urlencode($user)));
```
Comment (nickvergessen) [inline apps/files/lib/Controller/ViewController.php:269]:
```suggestion
		$nav->assign('webdav_url', \OCP\Util::linkToRemote('dav/files/' . rawurlencode($user)));
```

I think? Always forget which one was better

Comment (miaulalala) [reply apps/files/lib/Controller/ViewController.php:269]: I think rawurlencode:

[If you are encoding path segment, use rawurlencode().
If you are encoding query component, use urlencode().](https://www.gyrocode.com/articles/php-urlencode-vs-rawurlencode/)

**Review by szaimen — COMMENTED** [review]
@miaulalala can you add a teste for this?

**Review by nickvergessen — APPROVED** [review]

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
@miaulalala please squash

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**